ui/remoteAccess.js

```diff
@@ -0,0 +1,125 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+/* exported RemoteAccessApplet */
+/*
+ * Copyright © 2020 Endless OS Foundation LLC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { GObject, Meta } = imports.gi;
+
+const PanelMenu = imports.ui.panelMenu;
+const PopupMenu = imports.ui.popupMenu;
+
+/* HACK: The upstream impl hides the indicator on non wayland sessions
+ * and given we enable X11 on Endless OS atm let's enable it
+ * here. The original code was copied from gnome-shell 3.38 and
+ * we create a new class instead of specializing the upstream one
+ * to avoid some hacks that would be needed to handle object
+ * destruction properly.
+ */
+var RemoteAccessApplet = GObject.registerClass(
+class RemoteAccessApplet extends PanelMenu.SystemIndicator {
```

Comment:
I think you might want to use a different name here. Currently the GType name doesn't clash with Shell's because of the namespace, but given how critical a type clash is, I'd not risk. Something like `EndlessRemoteAccessApplet` should work.

Reply:
Ok, renamed to RemoteAccessIndicator instead - but as you said the namespace should protect us here - didn't use Endless as the indicator is not really endless specific like for example the EndlessButton.